### PR TITLE
Implement Power Diagram Generator

### PR DESCRIPTION
**Pull request by shineyruan**
This PR implements power diagram generator for distributed equitable power diagram partitioning algorithms.

See [Pavone's paper](https://stanfordasl.github.io/wp-content/papercite-data/pdf/Pavone.Arsie.ea.TAC11.pdf) for algorithm details.

### CODE REVIEW

**Comment by LejunJiang** [comment]
will review it on Friday night

File: src/Node.py [hunk 1]
```diff
@@ -50,9 +62,21 @@ def __init__(self, uid, mode='random'):
           'Current assignment mode is not supported: %s' % self.current_mode)
     ####################################
 
-    # Divide and Conquer
+    ################ Divide and Conquer ###############
     self.tsp_path = []
     self.past_tasks = []
+    self.done_partition = False  # self partitioning termination flag
```
Comment (LejunJiang) [inline src/Node.py]: Put those parameters into the if statement since only mode_dc uses them?

Comment (LejunJiang) [inline src/Node.py]: Put those parameters into the if statement since only mode_dc uses them?

**Review by LejunJiang — APPROVED** [review]

File: src/Node.py (hunk 1 above)
Comment (shineyruan) [reply src/Node.py]: Sure. Done!

**Pull request merged**